### ISSUE TITLE
Random pages in book view or Picture In Picture (PIP) feature

### ISSUE THREAD

**Issue by ElectronicSupersonic** [sumatrapdfreader/sumatrapdf#230]
Hi,
thanks for the PDF reader that just works (and does it as you want it to)!

I have a feature suggestion. The feature in somewhat similar to the Book View already implemented, but the difference would be in what pages can be displayed side by side.

As it is now, in the Book View display two pages side by side, these can only be two consequent
t pages (say page 10 with page 11). The feature I suggest would enable you to chose which pages of the document would be displayed side by side (say page 10 with page 23 side by side). Optionally it could be extended to display more then two pages side by side (for lucky ones with 32' displays :) ), say three or more. 

The idea behind it is that some PDF (or documents in  other formats) may have two related topics positioned on non consequent pages. For instance a drawing of a part and a table with dimensions of the part being placed on separate (non consequent) pages, or specification of a part and plots showing results of test measurements.

The same can be accomplished by using windows instead of tabs (i.e. open several instances of a document and view different pages of interest each in a separate window), but a combination of tabbed view and ability to view two or more different pages side by side in the same window IMO would be more convenient.

In other words it's a suggestion to modify/expand Book View feature with some added functionality.

Thanks!

**Comment by GitHubRulesOK** [comment]
There are hundreds of ways to set up multiple views 
here is one where the same document is opened two different scales but it could be different pages different user interface, spot the difference, whatever you want.
![image](https://user-images.githubusercontent.com/12754913/70589504-c96c9600-1bc7-11ea-8b1a-d282e57db3f4.png)

**Comment by SumatraPeter** [comment]
> ability to view two or more different pages side by side in the same window IMO would be more convenient

@kjk: This is a request for a split window feature, so you can consolidate with #918.

**Comment by GitHubRulesOK** [comment]
@kjk perhaps consolidate or close ?

**Comment by kjk** [comment]
Let's assume that multiple windows feature covers this https://forum.sumatrapdfreader.org/t/new-upcoming-3-2-feature-multiple-windows/2554

**Comment by ElectronicSupersonic** [comment]
@kjk 
This is not what I was suggesting. The idea was to be able to see two or more different pages of the SAME document in the SAME **TAB** (tab - not window). Multiple windows feature does what I'm already doing. I.E. in order to view multiple pages of the same document, I double click on the same document in explorer/double commander. This opens the same document in a different window (i.e. two or more instances of the application are running at the same time - instead of a single instance if implemented via tabs with a single tab showing different pages of the same document).
![SumatraPDF_new](https://user-images.githubusercontent.com/13113681/71468159-bd8af180-27cd-11ea-9f6b-4765730cb9e8.png)
vs
![SumatraPDF_old](https://user-images.githubusercontent.com/13113681/71468181-cc71a400-27cd-11ea-8605-1f89ea528ee6.png)

**Comment by kjk** [comment]
I understand but this achieves the same result, approximately, with 1/10 of the effort.

**Comment by ElectronicSupersonic** [comment]
> 
> 
> ... this achieves **the SAME** result, approximately, with 1/10 of the effort.
Well, sorry for being an annoyance, but it is not the same. Let me explain why with a real life example.

Say we need to select a part for an electronic device. Several parts appear to match the key requirements, now we need to chose one that is most suitable. In a datasheet key characteristics of each component are placed at different pages (**in non consecutive order** - read almost random pages). Each vendor has it's own way of arranging data in their datasheets, hence it would be a random set of pages for each datasheet.

Now in order to compare the three components we would need to open each page of interest in its own separate window. Say we need three pages from each document. That's 9 windows (3 windows per document x 3 documents). Now try to navigate between them all (with Alt+Tab) and not get confused. Sure, one can use multiple desktops (a separate desktop for each set of windows), but this feature is only available in Windows 10/Linux (which is not supported by the application). Windows 7 users are out of luck in this regard (one can argue that Windows 7 is soon to become a non supported OS, but I bet it will still be used for a while, plus my initial post was posted way back in Jun 2015 :) ). With each document **opened in its own tab** (_all the pages need from that particular document in that particular tab_) and **all the tabs in the same window**, it would be way more manageable and way less confusing. Plus it would be way easier to compare the documents that are on the same screen at the same time (albeit in different tabs).

Another example (somewhat related to the above example) is when one need to work with a PDF document in conjunction with some other application(s). Say you design a PCB (printed circuit board) for an electronic device in the dedicated application. As a part of the design workflow, you would need to use datasheets for all components used. Thus both an eCAD application and multiple PDF datasheets could be opened at once. It might not be a problem when multiple displays/monitors are used, but not everybody can afford multiple displays. It is a major pain in the *** to switch between the opened windows, especially when there are multiple datasheets opened.

Thanks!
